File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int c=0;
        int ele;
        for(int i=0;i<nums.size();i++){
            if(c==0){
                c=1;
                ele=nums[i];
            }
            else if(nums[i]==ele){
                c++;
            }
            else{
                c--;
            }
        }
        int cnt1=0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]==ele)
            cnt1++;
        }
        if(cnt1>nums.size()/2)
        return ele;
        return -1;
    }
    
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int count=0;
        int ele=-1;
        for(int i=0;i<nums.length;i++){
            if(count==0 ){
                ele=nums[i];
                count++;
            }
            else if(nums[i]==ele){
                count++;

            }
            else{
                count--;
            }
        }
        int c=0;
        for(int i=0;i<nums.length;i++){
            if(ele==nums[i]){
                c++;
            }
        }
        if(c>Math.ceil(nums.length/2))
        return ele;
        return -1;
    }
}